socs/agents/pysmurf_controller/agent.py

```diff
@@ -706,12 +717,83 @@ def take_bias_steps(self, session, params):
             bsa = bias_steps.take_bias_steps(
                 S, cfg, **params['kwargs']
             )
+
+            biased = np.logical_and.reduce([
+                params['rfrac_range'][0] < bsa.Rfrac,
+                params['rfrac_range'][1] > bsa.Rfrac
+            ])
             session.data = {
-                'filepath': bsa.filepath
+                'filepath': bsa.filepath,
+                'biased_total': int(np.sum(biased)),
+                'biased_per_bg': [
+                    int(np.sum(biased[bsa.bgmap == bg])) for bg in range(12)
+                ],
             }
+            quantiles = np.array([15, 25, 50, 75, 85])
+
+            def publish_quantile_block(arr, name):
+                if np.isnan(arr).all():
+                    return None
+                labels = [f'{name}_q{q}' for q in quantiles]
+                qs = [float(np.nanquantile(arr, q / 100)) for q in quantiles]
+                block = {
+                    k: q
+                    for k, q in zip(labels, qs)
+                }
+                count = int(np.sum(~np.isnan(arr)))
+                block[f'{name}_count'] = count
+
+                session.data[f'{name}_quantiles'] = {
+                    name: qs,
+                    'quantiles': labels,
+                    'count': count,
+                }
+
+                d = {
+                    'timestamp': time.time(),
+                    'block_name': f'{name}_quantile',
+                    'data': block
+                }
+
+                self.agent.publish_to_feed('bias_step_quantiles', d)
+                return d
+
+            # Resistance quantiles
+            d = publish_quantile_block(bsa.R0, 'Rtes')
+            d = publish_quantile_block(bsa.Si, 'resposnivity')
+            d = publish_quantile_block(bsa.Rfrac, 'Rfrac')
 
             return True, "Finished taking bias steps"
 
+    @ocs_agent.param('bgs', default=None)
+    @ocs_agent.param('kwargs', default=None)
+    def overbias_tes(self, session, params):
+        """overbias_tes(bgs=None, kwargs=None)
+
+        **Task** - Overbiases detectors using S.overbias_tes_all.
+
+        Args
+        -------
+        bgs : List[int]
+            List of bias groups to overbias. If this is set to None, it will
+            use all active bgs.
+        kwargs : dict
+            Additional kwargs to pass to the ``overbias_tes_all`` function.
+        """
+        kw = {'bias_groups': params['bgs']}
+        if params['kwargs'] is None:
+            kw.update(params['kwargs'])
```

Comment:
If `params['kwargs']` is None, won't this error out?

```
>>> a = {'test': 1}
>>> a.update(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
```

Meant to be `is not None`?

Reply:
Oops good catch